aten/src/ATen/ScalarOps.h

```diff
@@ -11,16 +12,16 @@ namespace c10 {
 // to implement this without going through Derived Types (which are not part of core).
 inline at::Tensor scalar_to_tensor(Scalar s, const Device device = at::kCPU) {
   // This is the fast track we have for CPU scalar tensors.
-  if (device == at::kCPU) {
+  if (device == at::kCPU && !s.isComplex()) {
     if (s.isFloatingPoint()) {
-      return at::native::scalar_tensor(s, at::device(at::kCPU).dtype(at::kDouble));
+      return at::scalar_tensor_fast(s, at::device(at::kCPU).dtype(at::kDouble));
     } else if (s.isBoolean()) {
-      return at::native::scalar_tensor(s, at::device(at::kCPU).dtype(at::kBool));
+      return at::scalar_tensor_fast(s, at::device(at::kCPU).dtype(at::kBool));
     } else if (s.isComplex()) {
```

Comment:
Q. @iseeyuan Maybe I'm missing something, but the conditions here seem to first exclude any scalar that is complex and then later checks if a complex scalar was found.

Reply:
Good catch. The later complex check part should be removed because it's not supported by fast pass.